students/cem/mailroom/mailroom_pt2.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
+import sys
+import datetime
+import os
+
+"""
+The script prompts the user (you) to choose from a menu of 
+3 actions: “Send a Thank You”, “Create a Report” or “quit”.
+"list" will bring up up donors list.
+"q" will quit the application.
+"""
+
+CHARITY_NAME = "Cem's Charity"
+
+donors = {
+    "Bill Gates": [653772.32, 12.17],
+    "Jeff Bezos": [877.33],
+    "Paul Allen": [663.23, 43.87, 1.32],
+    "Mark Zuckerberg": [1663.23, 4300.87, 10432.0],
+    "Thierry Henry": [6777.00, 8750.00],
+    "The Rock": [90000.00, 5550.00],
+    }
+
+prompt = "\n".join(("\nWelcome to the Mailbox!",
+                    "Please choose from below options:",
+                    "1 - Send a Thank You to a single donor",
+                    "2 - Create a Report",
+                    "3 - Send letters to all donors",
+                    "q - Quit",
+                    ">>> "))
+
+
+def list_donors():
+    """Prints a list of all donors in list"""
+    print("\nList of all {} donors: ".format(len(donors)), '\n', '-' * 25)
+    for donor in donors.keys():
+        print(f" ", donor)
+    print('-' * 25)
+
+
+def thank_you_email(response, donated):
+    """Prints out thank you letter"""
+    # for donor, value in donors.items():
+    letter = (
+        f"""
+        Dear {response}, 
+
+        Thank you for your amazingly generous donation of ${donated:.2f}! 
+        We, at {CHARITY_NAME}, greatly appreciate your donation, and your sacrifice. 
+
+        Your support helps to further our mission.
+
+        Regards,
+        Cem
+        """.format(**donors))
+    return letter
+
+
+def thank_you_email_all():
+    path = input(
+        f"\nEnter a dir name to create folder here in this pwd, full path or go back directories with '../'"
+        f"\nYour pwd now is {os.getcwd()}: ")
+    full_dir = os.path.join(os.getcwd() + "/" + path)
+    while True:
+        if path == 'q':
+            print("Quitting.")
+            break
+        else:
+            print(full_dir)
+            try:
+                os.mkdir(path)
+            except OSError:
```

Comment:
I like how you resolved this. I was a bit confusing, since the user is prompted to enter relative, not absolute path to a new folder, but by looking at code I found out what I was doing wrong.